docs/writing-tests/testharness-api.md

```diff
@@ -818,7 +818,14 @@ asserts that one `assert_func(actual, expected_array_N, extra_arg1, ..., extra_a
   allows multiple behaviours. Test authors should not use this method simply to hide
   UA bugs.
 
-## Formatting ##
+## Utility functions ##
+
+### `on_event(object, event, callback)`
```

Comment:
Can we mark this as deprecated or something? It seems like a largely pointless helper.

Reply:
I tend to agree, but that change reflects a subjective judgement, and I'd like to be able to discuss it separately. Would you merge this if I follow up with a patch to deprecate the function?

Reply:
That seems unnecessarily bureaucratic. I think the reason this wasn't documented in the first place was basically that it was clearly pointless but no one wanted to do the cleanup to actually remove it. So my first preference would be for that cleanup to happen and the function to be removed, my second preference would be to do nothing, my third preference would be to land these docs with it marked as deprecated and my last preference would be to land these docs without it being marked as deprecated, even if such a thing happened in a followup.

If the concern is that marking it as deprecated amounts to a RFC-requring change in a public API, then I think we'd be better off spending the energy on an RFC to remove it entirely, dropping this PR, and doing the largely mechanical updates to ~350 uses of the function [1]

[1] https://searchfox.org/mozilla-central/search?q=on_event(&case=false&regexp=false&path=web-platform%2Ftests

Reply:
I agree that the RFC process is a little heavy for marking something as deprecated. At the same time, I'm not comfortable making decisions about the public API based on feedback from one reviewer. We can use this pull request to increase our confidence that deprecation is generally acceptable, but in the mean time, the function will continue to be undocumented and untested.